src/meta/typesmap.go

```diff
@@ -173,6 +173,34 @@ func (m TypesMap) IsArray() bool {
 	return false
 }
 
+// HasOnlyArrays checks if map contains only arrays of any type
+func (m TypesMap) HasOnlyArrays() bool {
```

Comment:
These 2 new TypesMap predicates only used once. Why not use `m.Contains` with a callback there?
TypesMap already has quite a lot of such 1-time use (very ad-hoc) methods.
If you really want them, you can write them as util functions somewhere in linter package.
The simpler TypesMap API is, the easier it will be to refactor/replace it later.

Reply:
Done.
The functions were inlined.